packages/cli/src/commands/hydrogen/dev.ts

```diff
@@ -175,38 +221,44 @@ export async function runDev({
   async function safeStartMiniOxygen() {
     if (miniOxygen) return;
 
-    miniOxygen = await startMiniOxygen(
-      {
-        root,
-        debug,
-        assetsPort,
-        inspectorPort,
-        port: appPort,
-        watch: !liveReload,
-        buildPathWorkerFile,
-        buildPathClient,
-        env: await envPromise,
-      },
-      legacyRuntime,
-    );
-
-    enhanceH2Logs({host: miniOxygen.listeningAt, ...remixConfig});
+    const [tunnelHost, newMiniOxygen] = await Promise.all([
+      customerAccountPush && cliConfig
+        ? startTunnelAndPushConfig(root, cliConfig, appPort, storefront?.id)
+        : undefined,
```

Comment:
I bundle it together at first because the host display is in `safeStartMiniOxygen`, but I see your point and will try to figure out how to make tunnel start earlier.